packages/protocol/test/authorizers/Api3TokenLock.sol.js

```diff
@@ -0,0 +1,755 @@
+        context('caller is metaAdmin', async () => {
+          it('is set', async () => {
+            await api3TokenLock.connect(roles.metaAdmin).setLockAmount(NEW_LOCK_AMOUNT);
+
+            expect(await api3TokenLock.lockAmount()).to.equal(NEW_LOCK_AMOUNT);
+          });
+
+          it('emits event', async () => {
+            await expect(api3TokenLock.connect(roles.metaAdmin).setLockAmount(NEW_LOCK_AMOUNT))
+              .to.emit(api3TokenLock, 'SetLockAmount')
+              .withArgs(NEW_LOCK_AMOUNT);
+          });
+        });
+
+        context('caller is superAdmin', async () => {
+          it('is set', async () => {
+            await api3TokenLock.connect(roles.superAdmin).setLockAmount(NEW_LOCK_AMOUNT);
+
+            expect(await api3TokenLock.lockAmount()).to.equal(NEW_LOCK_AMOUNT);
+          });
+
+          it('emits event', async () => {
+            await expect(api3TokenLock.connect(roles.superAdmin).setLockAmount(NEW_LOCK_AMOUNT))
+              .to.emit(api3TokenLock, 'SetLockAmount')
+              .withArgs(NEW_LOCK_AMOUNT);
+          });
+        });
+
+        context('caller is admin', async () => {
+          it('is set', async () => {
+            await api3TokenLock.connect(roles.admin).setLockAmount(NEW_LOCK_AMOUNT);
+
+            expect(await api3TokenLock.lockAmount()).to.equal(NEW_LOCK_AMOUNT);
+          });
+
+          it('emits event', async () => {
+            await expect(api3TokenLock.connect(roles.admin).setLockAmount(NEW_LOCK_AMOUNT))
+              .to.emit(api3TokenLock, 'SetLockAmount')
+              .withArgs(NEW_LOCK_AMOUNT);
+          });
+        });
+
+        context('caller is not metaAdmin', async () => {
+          it('reverts', async () => {
+            await expect(api3TokenLock.connect(roles.unknown).setLockAmount(NEW_LOCK_AMOUNT)).to.revertedWith(
+              Errors.LowRank
+            );
+          });
+        });
+      });
+
+      context('Locked amount is zero', () => {
+        it('reverts', async () => {
+          await expect(api3TokenLock.connect(roles.metaAdmin).setLockAmount(0)).to.revertedWith(Errors.ZeroAmount);
+        });
+      });
+    });
+
+    describe('blockRequester', async () => {
+      context('caller is metaAdmin', async () => {
+        context('requester is blocking on an airnode', async () => {
+          it('sets the block', async () => {
+            await api3TokenLock.connect(roles.metaAdmin).blockRequester(airnodeAddress, roles.requester.address);
+
+            expect(
+              await api3TokenLock.airnodeToRequesterToBlockStatus(airnodeAddress, roles.requester.address)
+            ).to.equal(true);
+
+            await expect(
+              api3TokenLock.connect(roles.sponsor).lock(airnodeAddress, roles.requester.address)
+            ).to.revertedWith(Errors.RequesterBlocked);
+
+            await api3TokenLock.connect(roles.sponsor).lock(roles.unknown.address, roles.requester.address);
```

Comment:
A comment here explaning what should happen after calling the lock() function would be helpful.

Reply:
agreed, added the comments for better understanding